src/gromo/growing_module.py

```diff
@@ -1183,32 +1183,58 @@ def reset_computation(self) -> None:
     def delete_update(
         self,
         include_previous: bool = True,
+        include_output: bool = False,
     ) -> None:
         """
         Delete the update of the optimal added parameters.
 
+        By default, we do not delete the extended_input_layer of this layer because it
+        could be used by the next layer.
+
         Parameters
         ----------
         include_previous: bool
-            if True delete the update of the previous layer
+            if True delete the extended_output_layer of the previous layer
+        include_output: bool
+            if True delete the extended_input_layer of this layer,
+            warning: this does not delete the extended_input_layer of the next layer
         """
         self.optimal_delta_layer = None
-        self.extended_input_layer = None
-        self.extended_output_layer = None
         self.scaling_factor = 0.0  # type: ignore
         # this type problem is due to the use of the setter to change the scaling factor
         self.parameter_update_decrease = None
         self.eigenvalues_extension = None
         self._pre_activity = None
         self._input = None
 
-        if include_previous and self.previous_module is not None:
-            if isinstance(self.previous_module, GrowingModule):
-                self.previous_module.extended_output_layer = None
-            elif isinstance(self.previous_module, AdditionGrowingModule):
-                raise NotImplementedError  # TODO
+        if include_output:
+            self.extended_output_layer = None
+
+        if self.extended_input_layer is not None:
+            self.extended_input_layer = None
+            if include_previous and self.previous_module is not None:
+                if isinstance(self.previous_module, GrowingModule):
+                    self.previous_module.extended_output_layer = None
+                elif isinstance(self.previous_module, AdditionGrowingModule):
+                    raise NotImplementedError  # TODO
+                else:
+                    raise NotImplementedError
+            elif self.previous_module is not None:
+                warnings.warn(
+                    f"The extended_input_layer of {self.name} has been deleted."
+                    f"However, the extended_output_layer associated stored in the previous module"
+                    f"named {self.previous_module.name} has not been deleted."
+                    "This may lead to errors when using extended_forward.",
+                    UserWarning,
+                )
             else:
```

Comment:
```suggestion
        # delete extended_input_layer
        if self.extended_input_layer is not None:
            self.extended_input_layer = None
            
        # delete extended_output_layer
        if include_output:
            self.extended_output_layer = None

        # delete previous module extended_output_layer
        if self.extended_input_layer is not None:
            # normal behavior
            if include_previous and self.previous_module is not None:
                if isinstance(self.previous_module, GrowingModule):
                    self.previous_module.extended_output_layer = None
                elif isinstance(self.previous_module, AdditionGrowingModule):
                    raise NotImplementedError  # TODO
                else:
                    raise NotImplementedError
            
            # risky behavior
            if not include_previous and self.previous_module is not None:
                warnings.warn(
                    f"The extended_input_layer of {self.name} has been deleted."
                    f"However, the extended_output_layer associated stored in the previous module"
                    f"named {self.previous_module.name} has not been deleted."
                    "This may lead to errors when using extended_forward.",
                    UserWarning,
                )
            
            # incorrect behavior
            if include_previous and self.previous_module is None:
```